### PR TITLE
Aden 2537 ab test

### PR DESCRIPTION
**Pull request by jacekjursza**
Enable AB test for placement of the message for users that are using AdBlock.

### CODE REVIEW

File: extensions/wikia/AdEngine/js/recovery/message.js [hunk 1]
```diff
@@ -93,23 +104,30 @@ define('ext.wikia.adEngine.recovery.message', [
...
-		createMessage('right-rail').done(function (messageContainer) {
+		createMessage('right-rail', messageVariant).done(function (messageContainer) {
 			$('#WikiaRail').prepend(messageContainer);
 		});
 	}
 
 	function injectMessage() {
-		injectTopMessage();
-		injectRightRailMessage();
+		var group = win.Wikia.AbTest ? Wikia.AbTest.getGroup(abTestConfig.experimentName) : false;
```
Comment (fraszczakszymon) [inline extensions/wikia/AdEngine/js/recovery/message.js]: We have `wikia.abTest` AMD module for that and it will simplify this line

**Comment by fraszczakszymon** [comment]
LGTM

File: extensions/wikia/AdEngine/js/recovery/message.js [hunk 2]
```diff
@@ -68,10 +81,10 @@ define('ext.wikia.adEngine.recovery.message', [
 		});
 	}
 
-	function createMessage(uniqueClassName) {
+	function createMessage(uniqueClassName, messageVariant) {
 		return getAssets().then(function (assets) {
 			var template = assets.mustache[0],
-				text = mw.message('adengine-recovery-message-blocking-message-a').rawParams([
+				text = mw.message('adengine-recovery-message-blocking-message-'+messageVariant).rawParams([
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/recovery/message.js:88]: What do you think about a comments above the line `//adengine-recovery-message-blocking-message-a` and in next line `//adengine-recovery-message-blocking-message-b`?

File: extensions/wikia/AdEngine/js/recovery/message.js [hunk 3]
```diff
@@ -10,7 +10,8 @@ define('ext.wikia.adEngine.recovery.message', [
 	'wikia.location',
 	'wikia.log',
 	'wikia.mustache',
-	'wikia.window'
+	'wikia.window',
+	require.optional('wikia.abTest')
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/recovery/message.js]: As discussed today we can remove `require.optional()`.

File: extensions/wikia/AdEngine/js/recovery/message.js [hunk 4]
```diff
@@ -93,23 +106,32 @@ define('ext.wikia.adEngine.recovery.message', [
...
-		createMessage('right-rail').done(function (messageContainer) {
+		createMessage('right-rail', messageVariant).done(function (messageContainer) {
 			$('#WikiaRail').prepend(messageContainer);
 		});
 	}
 
 	function injectMessage() {
-		injectTopMessage();
-		injectRightRailMessage();
+		if (abTest) {
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/recovery/message.js]: If we remove `require.optional()` we can remove the condition :)

**Comment by nandy-andy** [comment]
Everything except b93d03c22441b9bac49288cadbb9800e59aa01c7 commit message looks good to me :laughing:

**Pull request merged**